File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        retList = [[]]        
        
        for num in nums:
            currLen = len(retList)

            for i in range(currLen):
                retList.append(list(retList[i]))
                retList[-1].append(num)
        
        return retList

File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int ret = 1;
        int len = nums.size();

        for (int i = 0; i < len; i++) {
            int num = nums[i];

            if ((num > 0) && (num <= len) && (num != i + 1)) {
                while (true) {
                    int temp = nums[num - 1];
                    //if (temp == num) num = 0;
                    nums[num - 1] = num;
                    if (temp <= 0 || temp > len || temp == num) break;
                    num = temp;
                }
            }
        }

        for (int i = 0; i < len; i++) {
            if (nums[i] != i + 1) {
                return i + 1;
            }
        }

        return len + 1;
    }
};